todo_app/tests/test_endtoend.py

```diff
@@ -0,0 +1,80 @@
+from logging import log
+import todo_app
+from todo_app.data import itemstatus
+from todo_app.data.itemstatus import ItemStatus
+from todo_app.model.viewModel import ViewModel
+from todo_app.data.item import Item
+import pytest
+from dotenv import load_dotenv, find_dotenv
+import requests,os
+
+from selenium import webdriver
+from threading import Thread
+
+import todo_app.app
+
+load_dotenv()
+
+
+def add_board():
+    response = requests.post(
+        url=f'https://api.trello.com/1/boards',
+        params={
+            'name': 'Selenium',
+            'key': os.environ['TRELLO_KEY'],
+            'token': os.environ['TRELLO_TOKEN']
+        }
+    )
+    return response.json()['id']
+
+
+def del_board(board_id):
+    requests.delete(
+        url=f'https://api.trello.com/1/boards/{board_id}',
+        params={
+            'key': os.environ['TRELLO_KEY'],
+            'token': os.environ['TRELLO_TOKEN']
+        }
+    )
+
+@pytest.fixture(scope='module')
+def app_with_temp_board():
+    # Create the new board & update the board id environment variable
+    board_id = add_board()
+    os.environ['TRELLO_BOARD'] = board_id
+
+    # construct the new application
+    application = todo_app.app.create_app()
+    application.boardId = board_id
+    # start the app in its own thread.
+    thread = Thread(target=lambda: application.run(use_reloader=False))
+    thread.daemon = True
+    thread.start()
+    yield application
+
+    # Tear Down
+    thread.join(1)
+    del_board(board_id)
+
+@pytest.fixture
+def driver():
+    driver = webdriver.Firefox(executable_path=r'D:\Users\Jordi\Documents\corndel\ex3\DevOps-Course-Starterv2\geckodriver.exe')
```

Comment:
Rather than requiring developers to each change the code in order to run the test, you should add geckodriver to your PATH and change this line to just
```suggestion
    driver = webdriver.Firefox()
```

This is not just to make dev easier, you also won't be able to get a CI pipeline to run these tests on a build server if the executable path is hardcoded like this.

Reply:
You add geckodriver.exe to the .gitignore file.
It is dependent on your OS and it's a dependency rather than something you manage yourself, so don't include it in your codebase.
Instead, mention it as a requirement in the README.md

But since it's already being tracked, you'll also need to run `git rm --cached geckodriver.exe` to remove it.